### PR TITLE
Feat/HNG 113 create related articles on dynamic blog page

### LINKED ISSUE #113: [FEATURE] Create and implement the API integration for Related Articles on the dynamic blog page
# **Description**
Implement the design for the related articles, make it responsive on both the web and mobile screens. Make use of reusable components for the related ariticles. Integrate the api to map out the all related articles in the blog details page

# **Acceptance Criteria**
- Pixel perfect implementaiton
- Responsive implementation
- Clean code
- Make use of reusable components where needed, make one if needed
- Follow the [contribution guidelines](https://github.com/hngprojects/hng_boilerplate_nextjs/blob/0acf613964ea064ae211a91d0a74d13938e4bc23/CONTRIBUTING.md) and [code of conduct](https://github.com/hngprojects/hng_boilerplate_nextjs/blob/0acf613964ea064ae211a91d0a74d13938e4bc23/CODE_OF_CONDUCT.md)

# **Requirements**
- Display three related articles
- Proper error handling
- Proper loading indicator
- None breaking api integration

# **Testing**
- Write test to ensure the page renders the required contents

# **Deadline**
Friday 11:59 pm WAT

# **Tasks**
- [ ] Implement all the sections in the design
- [ ] Responsiveness across all screens
- [ ] Make use of reusable components where needed
- [ ] Write test for the implementations
- [ ] Proper commit message

# **Images**
Frame 53126
Frame 1171275100

[Figma design](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=1632-11446&m=dev)

### PR DESCRIPTION
**Pull request by Kored28**
# Description

**Closes #113 **

# Changes proposed

## What were you told to do?
Implement the design for the related article in the dynamic blog detail page

[FEATURE] Create and implement the API integration for Related Articles on the dynamic blog page

## What did you do?

- I implemented all the sections in the design
- I made the component responsive
- I made use of reusable components where needed
- I wrote test for the implementations
- I made sure I used proper commit message

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos
Design
Screenshot (8)
Screenshot (9)

Implentation
Screenshot (6)
Screenshot (7)

Testing
Screenshot (11)
Screenshot (12)

Proof of test passed
Screenshot (16)

### CODE REVIEW

File: src/app/blog/[id]/RelatedArticle.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+  return (
+    <div className="flex w-[372px] flex-col justify-center gap-[24px] bg-[#FAFAFA] px-[0px] py-[16px] sm:w-full sm:px-[100px]">
+      <h1 className="ml-[17px] text-[28px] font-bold text-[#525252] md:ml-[0px]">
+        Related Articles
+      </h1>
+
+      {loading ? (
+        <div className="p-[0] sm:px-[100px]">
+          <p className="text-[24px] font-bold text-[#525252]">Loading...</p>
```
Comment (JulianaSau) [inline src/app/blog/[id]/RelatedArticle.tsx]: Dont use custom colors. Use those specified in the tailwind config

File: src/components/blog/BlogCard.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          className="absolute left-[15px] top-[15px] rounded-[16px] px-[17px] py-[5px]"
+        >
+          <p className="text-[12px] font-bold text-[#FAFAFA]">
+            {newsType.name}
+          </p>
+        </div>
+      </div>
+      <div className="flex h-[147px] w-[372px] flex-col gap-[24px] p-[16px] md:h-[137px] md:w-[354px]">
+        <p className="w-[320px] text-[20px] font-bold text-[#525252]">
+          The Power of Networking: How to Build Meaningful Connections
```
Comment (JulianaSau) [inline src/components/blog/BlogCard.tsx]: This value should be a prop

File: src/components/blog/BlogCard.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        </p>
+        <div className="flex items-center justify-between">
+          <p className="text-[14px] font-normal text-[#525252]">
+            July 12, 2024
```
Comment (JulianaSau) [inline src/components/blog/BlogCard.tsx]: This value shoudl be a prop

File: src/components/blog/BlogCard.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+          </p>
+          <div className="rounded-[16px] bg-[#F3F4F6] p-[8px]">
+            <p className="text-[14px] font-normal text-[#525252]">5 min read</p>
```
Comment (JulianaSau) [inline src/components/blog/BlogCard.tsx]: This value should be a prop

File: src/test/Button.test.tsx [hunk 5]
```diff
@@ -77,7 +73,6 @@ describe("custom Button Component", () => {
 
   it("supports keyboard navigation and screen reader compatibility", async () => {
     expect.hasAssertions();
-
```
Comment (JulianaSau) [inline src/test/Button.test.tsx]: where are these changes coming from?

File: tsconfig.json [hunk 6]
```diff
@@ -34,7 +34,8 @@
     "next-env.d.ts",
     "**/*.ts",
     "**/*.tsx",
-    ".next/types/**/*.ts"
+    ".next/types/**/*.ts",
+    "**/**/*.tsx"
```
Comment (JulianaSau) [inline tsconfig.json]: what is this change for?

File: tailwind.config.ts [hunk 7]
```diff
@@ -113,8 +113,7 @@ const config = {
       },
     },
   },
-  // eslint-disable-next-line unicorn/prefer-module
   plugins: [require("tailwindcss-animate")],
 } satisfies Config;
 
-export default config;
+export default config;
```
Comment (JulianaSau) [inline tailwind.config.ts]: why are you adding a new line here as a change?

File: src/components/blog/BlogCard.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+          height={247}
+          className="h-[247px] w-[374px] rounded-t-lg"
+        />
+        <div
+          data-testid="news-displayed"
+          key={newsType.id}
+          style={{ backgroundColor: newsType.color }}
+          className="absolute left-[15px] top-[15px] rounded-[16px] px-[17px] py-[5px]"
+        >
+          <p className="text-[12px] font-bold text-[#FAFAFA]">
```
Comment (JulianaSau) [inline src/components/blog/BlogCard.tsx]: Dont use custom colors. Use the ones in the tailwind config

**Review by JulianaSau — CHANGES_REQUESTED** [review]

File: tailwind.config.ts (hunk 7 above)
Comment (Kored28) [reply tailwind.config.ts]: It's because I the dev has been updated

File: src/test/Button.test.tsx (hunk 5 above)
Comment (Kored28) [reply src/test/Button.test.tsx]: The main branch has been updated

File: src/app/blog/[id]/RelatedArticle.tsx (hunk 1 above)
Comment (Kored28) [reply src/app/blog/[id]/RelatedArticle.tsx]: okay

Comment (Kored28) [reply src/app/blog/[id]/RelatedArticle.tsx]: There is no one specified for blogs

**Pull request closed without merging**